Gems/ROS2/Code/Source/VehicleDynamics/DriveModels/AckermannDriveModel.cpp

```diff
@@ -0,0 +1,168 @@
+{
+    void AckermannDriveModel::Reflect(AZ::ReflectContext* context)
+    {
+        PidConfiguration::Reflect(context);
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<AckermannDriveModel, DriveModel>()
+                ->Version(1)
+                ->Field("SteeringPID", &AckermannDriveModel::m_steeringPid)
+                ->Field("SpeedPID", &AckermannDriveModel::m_speedPid);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<AckermannDriveModel>("Simplified Drive Model", "Configuration of a simplified vehicle dynamics drive model")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &AckermannDriveModel::m_steeringPid,
+                        "Steering PID",
+                        "Configuration of steering PID controller")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::Default,
+                        &AckermannDriveModel::m_speedPid,
+                        "Speed PID",
+                        "Configuration of speed PID controller");
+            }
+        }
+    }
+
+    void AckermannDriveModel::Activate(const VehicleConfiguration& vehicleConfig)
+    {
+        m_driveWheelsData.clear();
+        m_steeringData.clear();
+        m_vehicleConfiguration = vehicleConfig;
+        m_speedPid.InitializePid();
+        m_steeringPid.InitializePid();
+    }
+
+    void AckermannDriveModel::ApplyInputState(const VehicleInputsState& inputs, uint64_t deltaTimeNs)
+    {
+        if (m_driveWheelsData.empty())
+        {
+            m_driveWheelsData = VehicleDynamics::Utilities::GetAllDriveWheelsData(m_vehicleConfiguration);
+        }
+
+        if (m_steeringData.empty())
+        {
+            m_steeringData = VehicleDynamics::Utilities::GetAllSteeringEntitiesData(m_vehicleConfiguration);
+        }
+
+        ApplySteering(inputs.m_steering.GetValue(), deltaTimeNs);
+        ApplySpeed(inputs.m_speed.GetValue(), deltaTimeNs);
+    }
+
+    void AckermannDriveModel::ApplyWheelSteering(SteeringDynamicsData& wheelData, float steering, double deltaTimeNs)
+    {
+        const double deltaTimeSec = double(deltaTimeNs) / 1e9;
+
+        const auto& steeringEntity = wheelData.m_steeringEntity;
+        AZ::Vector3 currentSteeringElementRotation;
+        AZ::TransformBus::EventResult(currentSteeringElementRotation, steeringEntity, &AZ::TransformBus::Events::GetLocalRotation);
+        const float currentSteeringAngle = currentSteeringElementRotation.Dot(wheelData.m_turnAxis);
+        const double pidCommand = m_steeringPid.ComputeCommand(steering - currentSteeringAngle, deltaTimeNs);
+        if (AZ::IsClose(pidCommand, 0.0)) // TODO - use the third argument with some reasonable value which means "close enough"
```

Comment:
This seems like a constant for epsilon in this function should be added and the TODO removed.